test/suite/test_import08.py

```diff
@@ -0,0 +1,154 @@
+        cursor2 = session2.open_cursor('metadata:')
+        generated_uri = None
+        for k, _ in cursor2:
+            if k.startswith('table:'):
+                generated_uri = k
+                break
+        cursor2.close()
+
+        # Now begin a transaction and remove the first entry in the table.
+        #
+        # This is going to allocate a transaction ID and more importantly, will ensure that pinned
+        # ID is set for subsequent checkpoints in this test. If we don't do this, reconciliation
+        # won't actually write transaction IDs to the disk since they will be deemed obsolete.
+        session2.begin_transaction()
+        self.assertTrue(generated_uri is not None)
+        cursor2 = session2.open_cursor(generated_uri)
+        cursor2.set_key(0)
+        cursor2.remove()
+
+        # Now create the file that we'll be importing later.
+        self.session.create(self.uri, self.create_config)
+
+        # Insert records into our newly created file.
+        #
+        # Since we allocated a bunch of transaction IDs before, we'll be allocating pretty high
+        # IDs at this point.
+        for i in range(0, len(self.keys)):
+            self.update(self.uri, self.keys[i], self.values[i], self.ts[i])
+
+            # We want to checkpoint after each write to increase the write gen for this particular
+            # btree.
+            self.session.checkpoint()
+
+        # Export the metadata for the table.
+        c = self.session.open_cursor('metadata:', None, None)
+        original_db_file_config = c[self.uri]
+        c.close()
+
+        self.printVerbose(3, '\nFile configuration:\n' + original_db_file_config)
+
+        # Now that we've finished doing our checkpoints, we can let go of the transaction ID we
+        # allocated earlier.
+        session2.rollback_transaction()
+
+        # Close the connection.
+        self.close_conn()
+
+        # Create a new database and connect to it.
+        newdir = 'IMPORT_DB'
+        shutil.rmtree(newdir, ignore_errors=True)
+        os.mkdir(newdir)
+        self.conn = self.setUpConnectionOpen(newdir)
+        self.session = self.setUpSessionOpen(self.conn)
+
+        # Bring forward the oldest to be past or equal to the timestamps we'll be importing.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(self.ts[-1]))
+
+        # Copy over the datafiles for the object we want to import.
+        self.copy_file(self.original_db_file, '.', newdir)
+
+        # Contruct the config string.
+        if self.repair:
+            import_config = 'import=(enabled,repair=true)'
+        else:
+            import_config = 'import=(enabled,repair=false,file_metadata=(' + \
+                original_db_file_config + '))'
+
+        # Import the file.
+        self.session.create(self.uri, import_config)
+
+        # Verify object.
+        self.session.verify(self.uri)
+
+        # Check that the values are all visible.
```

Comment:
Do you think we should create a new table in the new database and check that the write gens in the metadata are what we expect for both the new and imported table?

Reply:
Not sure. I think the expected write generation should be deterministic and consistent but I think the number will look quite random (when I was debugging, it was 18 or something like that). I can give it a try though.

Reply:
7da9f81

Reply:
7612ad1